src/samples.jl

```diff
@@ -117,5 +117,98 @@ function _sample_unit_nball_muller!(D::Vector{Vector{N}}, n::Int, p::Int;
     return D
 end
 
+## Rejection Sampling
+
+"""
+    _canonical_length(X::LazySet)
+
+Get the canonical length of the box-approximation of the convex set `X`.
+
+### Inputs
+
+- `X` - convex set
+
+### Outputs
+
+A matrix with `n = dims(X)` rows and two columns. Each row stands for
+one dimension of `X` whereas the first column is the minimum and the second
+column is the maximum value of the corresponding dimension.
+"""
+function _canonical_length(X::LazySet{N}) where {N<:Real}
+    dims = dim(X)
+    x = Matrix{N}(undef, dims, 2)
+    for j = 1:dims
+        ej = SingleEntryVector(j, dims, one(N))
+        x[j,:] = [-ρ(-ej, X), ρ(ej, X)]
+    end
+    return x
+end
+
+"""
+    rand(X::LazySet{N}, n_samples::Int;
+           rng::AbstractRNG=GLOBAL_RNG,
+           seed::Union{Int, Nothing}=nothing) where {N}
+
+Rejection sampling of an arbitrary LazySet `X` for which the support value
+function is defined.
+
+Draw a sample `x` from a uniform distribution of a box-overapproximation of the
+original set `X` in all `n` dimensions. The function rejects a drawn sample `x`
+and redraws as long as the sample is not contained in the original set `X`,
+i.e., `x ∉ X`.
+
+### Input
+
+- `X`           -- lazyset
+- `n_samples`    -- number of random samples
+- `rng`         -- (optional, default: `GLOBAL_RNG`) random number generator
+- `seed`        -- (optional, default: `nothing`) seed for reseeding
+
+### Output
+
+A vector of `n_samples` vectors.
+"""
+function sample(X::LazySet{N}, n_samples::Int;
+                rng::AbstractRNG=GLOBAL_RNG,
+                seed::Union{Int, Nothing}=nothing) where {N<:Real}
+    @assert isbounded(X) "this function requires that the set `X` is bounded, but it is not"
+
+    D = Vector{Vector{N}}(undef, n_samples) # preallocate output
+    sample!(D, SetSampler(X); rng=rng, seed=seed)
+    return D
+end
+
+function sample(X::LazySet{N}; kwargs...) where {N<:Real}
```

Comment:
This is ok but i don't see a compelling reason to not give the default value, 1, to the function defined in line 171. The second argument can still have a default value without being a keyword argument. What do you think?

Reply:
I agree with you in general. But I did this to avoid having a `Vector{Vector}` as a result when I only want one sample from a set. Therefore the `[1]` in `sample(X, 1; kwargs...)[1]`. Is there a better way to handle this?

Reply:
I disagree. Type instabilities are not good.

Reply:
You can add the default value `1` to the other method, but the return type should *always* be a vector of vectors. For convenience it is useful to have this other method that only returns a single vector instead.

Reply:
True, then forget about the default value `1`.

Reply:
I misunderstood your comment completely, sorry. I agree with what you said :+1: